.github/workflows/simulation_test.yml

```diff
@@ -55,23 +77,27 @@ jobs:
 
     - name: Install NEURON
       run: |
-        sudo apt-get install flex libfl-dev bison ninja-build
-        pip install -U pip setuptools
-        pip install "cython<3" pytest sympy
         export SONATAREPORT_DIR=$(pwd)/libsonatareport/build/install
+        git clone https://github.com/BlueBrain/nmodl.git --depth=1
         git clone --branch="${NEURON_BRANCH:-master}" https://github.com/neuronsimulator/nrn.git --depth=1
-        cd nrn
-        mkdir build && cd build
-        cmake -G Ninja -DPYTHON_EXECUTABLE=$(which python) -DCMAKE_INSTALL_PREFIX=$(pwd)/install -DNRN_ENABLE_MPI=ON -DNRN_ENABLE_INTERVIEWS=OFF \
-          -DNRN_ENABLE_CORENEURON=ON -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCORENRN_ENABLE_REPORTING=ON -DCMAKE_PREFIX_PATH=$SONATAREPORT_DIR ..
-        cmake --build . --parallel
-        cmake --build . --target install
+        python -m pip install --upgrade pip -r nrn/nrn_requirements.txt
+        python -m pip install --upgrade -r nmodl/requirements.txt
+        cmake -B nrn/build -S nrn -G Ninja \
+          -DPYTHON_EXECUTABLE=$(which python) \
+          -DCMAKE_INSTALL_PREFIX=$(pwd)/nrn/build/install \
+          -DNRN_ENABLE_MPI=ON \
+          -DNRN_ENABLE_INTERVIEWS=OFF \
+          -DNRN_ENABLE_CORENEURON=ON \
+          -DCMAKE_C_COMPILER=gcc \
+          -DCMAKE_CXX_COMPILER=g++ \
+          -DCORENRN_ENABLE_REPORTING=ON \
+          -DCMAKE_PREFIX_PATH=$SONATAREPORT_DIR
+        cmake --build nrn/build --parallel
+        cmake --build nrn/build --target install
```

Comment:
Can we also cache the `nrn` directory? Maybe pin a commit as the key?

It seems that neurodamus should not be _that_ reliant on NRN master. Caching `libsonata`, `libsonatareport` and `nrn` builds could save 7-10 minutes build time and conserve runner minutes.

Reply:
Yes, caching neuron will speed up a lot. It will be nice to cache the same commit as the neuron spack recipe but we can't know the ID before launching the workflow, right? Then we need to hardcode a version and update it manually?

Reply:
Sure. I will add the caching in a separate PR https://github.com/BlueBrain/neurodamus/pull/166 .